consensus/consensus_v2.go

```diff
@@ -290,16 +311,21 @@ func (consensus *Consensus) onAnnounce(msg *msg_pb.Message) {
 
 // tryPrepare will try to send prepare message
 func (consensus *Consensus) prepare() {
-	// Construct and send prepare message
-	msgToSend := consensus.constructPrepareMessage()
-	// TODO: this will not return immediatey, may block
 
-	if err := consensus.msgSender.SendWithoutRetry([]nodeconfig.GroupID{nodeconfig.NewGroupIDByShardID(nodeconfig.ShardID(consensus.ShardID))}, host.ConstructP2pMessage(byte(17), msgToSend)); err != nil {
-		utils.Logger().Warn().Err(err).Msg("[OnAnnounce] Cannot send prepare message")
-	} else {
-		utils.Logger().Info().
-			Str("blockHash", hex.EncodeToString(consensus.blockHash[:])).
-			Msg("[OnAnnounce] Sent Prepare Message!!")
+	for i, key := range consensus.CurrentPubKeys.PublicKey {
+		// Construct and send prepare message
+		msgToSend := consensus.constructPrepareMessage(key, consensus.CurrentPriKeys.PrivateKey[i])
+
+		go func(msgToSend []byte) {
+			if err := consensus.msgSender.SendWithoutRetry([]nodeconfig.GroupID{nodeconfig.NewGroupIDByShardID(nodeconfig.ShardID(consensus.ShardID))}, host.ConstructP2pMessage(byte(17), msgToSend)); err != nil {
```

Comment:
is it necessary to make this over and over? 

[]nodeconfig.GroupID{nodeconfig.NewGroupIDByShardID(nodeconfig.ShardID(consensus.ShardID))},

Reply:
It's not necessary. Changed it.